app/src/ui/repositories-list/index.tsx

```diff
@@ -65,7 +67,14 @@ export class RepositoriesList extends React.Component<IRepositoriesListProps, vo
   }
 
   public render() {
+    if (this.props.repositories.length > 0) {
+      this.hadRepos = true
+    }
+
     if (this.props.repositories.length < 1) {
+      if (this.hadRepos) {
+        this.props.dispatcher.closeFoldout(FoldoutType.Repository)
```

Comment:
Makes perfect sense. Did you see my comment on #1793? I think it may change how we go about this fix